#!/usr/bin/exec-suid --real -- /usr/bin/python -I
import sys
import importlib.util
from os.path import exists, join

def import_module_by_path(module_path):
   module_name = "custom_module"
   spec = importlib.util.spec_from_file_location(module_name, module_path)
   custom_module = importlib.util.module_from_spec(spec)
   spec.loader.exec_module(custom_module)
   return custom_module

try:
	USER_FILE = join('/home/hacker', sys.argv[1])
except IndexError:
	print("Please give me tha name of your python file like so\n")
	print("/challenge/verify <your_python_file_name_here>")
	exit(-1)

if not exists(USER_FILE):
	print("Error: That file doesn't seem to exist.")
	exit(-1)

try:
	emp = import_module_by_path(USER_FILE)
except ImportError as e:
	print(f"Failed to import data from {USER_FILE}.")
	print(e)
	exit(-1)

try:
	doc = emp.Doctor("Craig", 80987, 400000, "Pediatrician")
except Exception as e:
	print(e)
	print("I wasn't able to create an instance of your Doctor subclass. Make sure it's written correctly!")
	exit(0)

try:
	doc.display_info()
except Exception as e:
	print(e)
	print("Failed while calling 'display_info'. Make sure your 'display_info' method is correct!")
	exit(0)

try:
	doc.diagnose("Bailey", "Mumps")
except Exception as e:
	print(e)
	print("Failed while calling 'diagnose'. Make sure your 'diagnose' method is correct.")
	exit(0)

with open('/flag', 'r') as fObj:
	print(fObj.read())
